organization-menu-list-item-cy-217

steiningermona opened the pull request.

steiningermona committed:
finish text and image preview, WIP icon preview

steiningermona committed:
finish text and image preview, WIP icon preview

steiningermona committed:
finish previews

steiningermona committed:
finish previews

Commit:
fix: dev build has circular dep 🐛 (#76)

fix: dev build has circular dep 🐛

steiningermona committed:
WIP add badge component

radicaa committed:
CY-170-solid-label (#51)

* feat: started adding solid labels, styles are not done yet

* refactor: added styles

* fix: error

* refactor: CSSColors, Label styles

* feat: added to comp index and added index to solidLabel

* fix: styles should not only apply to p and span

steiningermona committed:
add story

steiningermona committed:
add story

steiningermona committed:
implement badge component WIP change background color based on value

steiningermona committed:
implement badge component WIP change background color based on value

steiningermona committed:
change color variable based on value

steiningermona committed:
change color variable based on value

steiningermona committed:
Merge branch 'dev' of https://github.com/caisy-io/league into badges-cy-109

steiningermona committed:
Merge branch 'badges-cy-109' into organization-menu-list-item-cy-217

steiningermona committed:
WIP commit

steiningermona committed:
WIP commit

steiningermona committed:
update styling to center badge

steiningermona committed:
update styling to center badge

steiningermona committed:
Merge branch 'badges-cy-109' into organization-menu-list-item-cy-217

steiningermona committed:
assemble component and implement conditional renderings

steiningermona committed:
assemble component and implement conditional renderings

steiningermona committed:
update story and small fixes

steiningermona committed:
update story and small fixes

radicaa committed:
Cy 84 layouts (#29)

* refactor: created new layout files, renamed old layout to layoutLegacy

* wip: resizable sider

* fix: resizable behaviour

* refactor: added license

* refactor: added interface, removed unnecessary useState

* refactor: added SLayoutSiderContent

* feat: added movable

* feat: added movable story

* feat: addes resizable story

* refactor: removed layout styles from resizable

* feat: resizable now possible for left and right in story

* refactor: removed unused imports

* feat: layout, sider bug

* fix: sider resize within contentwrapper

* fix: smoother resize bc of adding attrs

* refactor: added interface for fix height of top nav

* fix: behaviour when resizing with flex-grow, refactored components

* feat: added sidebar tags, sidebar can be opened/closed,...

* fix: scrolling, button stories and button position when scrolling

* fix: layout stories with one sider

* refactor: removed done todo

* fix: resizable stories works again

* merge: merged dev

* feat: added stackable

* feat: added js files

* Update package.json

remove strange deps

* refactor: make formatting the same as on dev again

* refactor: removed console log

* chore: remove unused and silly dependencies form

chore: remove unused and silly dependencies form package

* refactor: removed sidebarTag from buttons, added sidebarTag component

* added popover, needs to be replaced in SPopover

* refactor: removed puppeteer

* fix: yarn tsc should work now

* refactor: removed sidebar tag styles , etc from button

* chore: created example story to point out

chore: created example story to point out the error

* chore: lint

chore: lint

* fix: import issue

* refactor: target is displayed now, but still not correct

* refactor: added styles, placement is still not correct

* fix: stackable, triangle is shown now, poppable is weird near borders,...

* fix: behaviour near borders, poppable work snow in stories example

* refactor: removed logs, fixed formatting

* fix: popconfirm works again, triangle color is not correct

* fix: triangle clor

* fix: stackable import in drawer

* refactor: removed webrix and imports

* fix: storybok should work again

* chore: update the style of handing down

chore: update the style of handing down the props

* chore: change the button back to dev

chore: change the button back to dev

* chore: remove dup button storie

chore: remove dup button storie

* chore: added content to layout story

chore: added content to layout story

* chore: replace classnames

chore: replace classnames

* refactor: added dynamic content to left/right sider stories

* refactor: added webrix hooks, replaced them in datePricker and selectSingle, added stories

* refactor: replaced hooks in select single and multiple, added story

* refactor: changed some imports

* refactor: replaced hooks in search, added story

* refactor: hooks export and imports

* refactor: replaced hooks in table

* refactor: changed exports to alphabetical order

* refactor: replaced classnames

* refactor: opposite slider width is not needed for content width

* fix: classname bug, triangle did not render sometimes

* refactor: imports

* refactor: removed webrix

* refactor: removed empty styled component STriangle

* fix: linter issues

* refactor: imports

* refator: imports

* refactor: imports, bc of circular dependency

* fix: TPlacement export

* refactor: solved some TODOs and removed them

* fix: type in resizable

* Update SResizableWrapper.tsx

* fix: the missing number functions 🐛

fix: the missing number functions 🐛

Co-authored-by: Moritz Gruber <m.gruber@cyclus.digital>

steiningermona committed:
Merge branch 'dev' of https://github.com/caisy-io/league into badges-cy-109

steiningermona committed:
remove resize function

steiningermona committed:
update styling of badge to new design

steiningermona committed:
add position relative in center mode

steiningermona committed:
update-the-json-tokens-to-latest-from-cy-433 (#64)

* add new json, generate new tokens and update import in search bar operand to avoid build fail

* replace json

* fixed sd.build file to work with new typography tokens

Co-authored-by: Nicolas Shiken <nicolasshiken@gmail.com>

Commit:
make-storybook-fail-in-the-pipeline-if-cy-434 (#66)

* ci: update ci pipeline to fail if

ci: update ci pipeline to fail if storybooks print failed 🐳

* chore: update ignore again

chore: update ignore again

* fix: the script and move it to

fix: the script and move it to file 🐛

* chore: change i did not want

chore: change i did not want

* ci: update pipeline 🐳

ci: update pipeline 🐳

* ci: update ci order 🐳

ci: update ci order 🐳

* ci: create cache dir 🐳

ci: create cache dir 🐳

* ci: create cache dir 🐳

ci: create cache dir 🐳

* ci: update 🐳

ci: update 🐳

* ci: use sh 🐳

ci: use sh 🐳

* chore: remove forced out

chore: remove forced out

nicolasshiken committed:
Update SButton.tsx (#88)

nicolasshiken committed:
feat: we now generate the font-weights as numbers (#71)

diegomart2000 reviewed (changes requested):
@steiningermona please rebase from dev.

nicolasshiken committed:
chore: we now use the Flag Component instead of an SVG (#82)

Commit:
CY-84-layouts-apply-review-feedback (#84)

* refactor: created new layout files, renamed old layout to layoutLegacy

* wip: resizable sider

* fix: resizable behaviour

* refactor: added license

* refactor: added interface, removed unnecessary useState

* refactor: added SLayoutSiderContent

* feat: added movable

* feat: added movable story

* feat: addes resizable story

* refactor: removed layout styles from resizable

* feat: resizable now possible for left and right in story

* refactor: removed unused imports

* feat: layout, sider bug

* fix: sider resize within contentwrapper

* fix: smoother resize bc of adding attrs

* refactor: added interface for fix height of top nav

* fix: behaviour when resizing with flex-grow, refactored components

* feat: added sidebar tags, sidebar can be opened/closed,...

* fix: scrolling, button stories and button position when scrolling

* fix: layout stories with one sider

* refactor: removed done todo

* fix: resizable stories works again

* merge: merged dev

* feat: added stackable

* feat: added js files

* Update package.json

remove strange deps

* refactor: make formatting the same as on dev again

* refactor: removed console log

* chore: remove unused and silly dependencies form

chore: remove unused and silly dependencies form package

* refactor: removed sidebarTag from buttons, added sidebarTag component

* added popover, needs to be replaced in SPopover

* refactor: removed puppeteer

* fix: yarn tsc should work now

* refactor: removed sidebar tag styles , etc from button

* chore: created example story to point out

chore: created example story to point out the error

* chore: lint

chore: lint

* fix: import issue

* refactor: target is displayed now, but still not correct

* refactor: added styles, placement is still not correct

* fix: stackable, triangle is shown now, poppable is weird near borders,...

* fix: behaviour near borders, poppable work snow in stories example

* refactor: removed logs, fixed formatting

* fix: popconfirm works again, triangle color is not correct

* fix: triangle clor

* fix: stackable import in drawer

* refactor: removed webrix and imports

* fix: storybok should work again

* chore: update the style of handing down

chore: update the style of handing down the props

* chore: change the button back to dev

chore: change the button back to dev

* chore: remove dup button storie

chore: remove dup button storie

* chore: added content to layout story

chore: added content to layout story

* chore: replace classnames

chore: replace classnames

* refactor: added dynamic content to left/right sider stories

* refactor: added webrix hooks, replaced them in datePricker and selectSingle, added stories

* refactor: replaced hooks in select single and multiple, added story

* refactor: changed some imports

* refactor: replaced hooks in search, added story

* refactor: hooks export and imports

* refactor: replaced hooks in table

* refactor: changed exports to alphabetical order

* refactor: replaced classnames

* refactor: opposite slider width is not needed for content width

* fix: classname bug, triangle did not render sometimes

* refactor: imports

* refactor: removed webrix

* refactor: removed empty styled component STriangle

* fix: linter issues

* refactor: imports

* refator: imports

* refactor: imports, bc of circular dependency

* fix: TPlacement export

* refactor: solved some TODOs and removed them

* fix: type in resizable

* Update SResizableWrapper.tsx

* fix: the missing number functions 🐛

fix: the missing number functions 🐛

* chore: update layout default styles

chore: update layout default styles

* chore: refactord the story styles

chore: refactord the story styles

Co-authored-by: Andrijana Radic <a.radic@cyclus.digital>

steiningermona committed:
Merge branch 'organization-menu-list-item-cy-217' of https://github.com/caisy-io/league into organization-menu-list-item-cy-217

steiningermona committed:
Merge branch 'previews-cy-103' of https://github.com/caisy-io/league into previews-cy-103

steiningermona committed:
destructure props, only pass required props and switch to named import

steiningermona committed:
add size interface and interface folder

steiningermona committed:
fix: rename default size interface

steiningermona committed:
use interface in badge component

nicolasshiken committed:
created interfaces/index file and exported it

steiningermona committed:
solve merge conflicts with dev

Commit:
chore: clean up flag duplication

chore: clean up flag duplication

steiningermona committed:
solve merge conflict

Commit:
fix: hm turns out preview-flag-round does equal

fix: hm turns out preview-flag-round does equal flag and that can also be used for flag-rounded 🐛

Commit:
Merge remote-tracking branch 'origin/dev' into previews-cy-103

steiningermona committed:
Merge remote-tracking branch 'origin/previews-cy-103' into organization-menu-list-item-cy-217

diegomart2000 commented on src/components/badge/Badge.tsx:
```diff
@@ -1,40 +1,59 @@
 }
 
-export const Badge: React.FC<IBadge> = ({ ...props }) => {
+export const Badge: React.FC<IBadgeProps> = ({ children, ...props }) => {
```
Prefer destructuring and named imports for react. Also badge is not working properly. Test here https://league-pxa6boiva-caisy.vercel.app/?path=/story/components-badge--position-left&args=type:color;value:11

diegomart2000 commented on src/components/organization-menu-list-item/OrganizationMenuListItem.tsx:
```diff
@@ -0,0 +1,49 @@
+  height: 100%;
+`;
+
+export const OrganizationMenuListItem: React.FC<IOrganizationMenuListItemProps> = ({ ...props }) => {
```
Prefer destructuring and named imports. Only pass down used properties. Make sure component won't break on larger pieces of content.

https://league-pxa6boiva-caisy.vercel.app/?path=/story/components-organizationmenulistitem--default&args=title:Label+kkkk+dsdf+dsfsdfw+wer+wer+werwe+rwer+we+rwer;label:Label+kkkk+dsdf+dsfsdfw+wer+wer+werwe+rwer+we+rwer;size:small

diegomart2000 commented on src/components/preview-flag-round/PreviewFlagRound.tsx:
```diff
@@ -0,0 +1,13 @@
+import Flag from "../flag";
+import { IFlag } from "../flag/IFlag";
+
+const PreviewFlagRound: FC<IFlag> = ({ countryCode }) => {
```
This component is repeated and not actually needed, let's discuss on why the original is not being used.

diegomart2000 commented on src/components/preview-icon/PreviewIcon.stories.tsx:
```diff
@@ -0,0 +1,56 @@
+  },
+};
+
+function PreviewIconDemo(args) {
```
This component is repeated and not actually needed, let's discuss on why the original is not being used.

diegomart2000 commented on src/components/preview-icon/PreviewIcon.tsx:
```diff
@@ -0,0 +1,15 @@
+  size: number;
+}
+
+export const PreviewIcon: React.FC<IPreviewIcon> = ({ icon, ...props }) => {
```
This component is repeated and not actually needed, let's discuss on why the original is not being used.

steiningermona committed:
Merge branch 'badges-cy-109' into organization-menu-list-item-cy-217

steiningermona committed:
Merge remote-tracking branch 'origin/dev' into organization-menu-list-item-cy-217

steiningermona committed:
Merge branch 'badges-cy-109' into organization-menu-list-item-cy-217

steiningermona committed:
descturcure props and cleanup

steiningermona committed:
adjust border and colour value

steiningermona committed:
remove dotted input

steiningermona committed:
export interface in preview component and update list item to use preview component

steiningermona committed:
Merge branch 'dev' into organization-menu-list-item-cy-217

steiningermona committed:
solve merge conflict

steiningermona committed:
Merge remote-tracking branch 'origin/dev' into badges-cy-109

steiningermona committed:
forward ref and add to interface

steiningermona committed:
add ref to component

steiningermona committed:
delete unused components

steiningermona committed:
solve merge conflict

steiningermona commented:
**from figma**
![image](https://user-images.githubusercontent.com/93647258/145040973-bc30c551-3872-4a94-8e5f-6921f43fd72e.png)
**from storybook**
- small/ medium
![image](https://user-images.githubusercontent.com/93647258/145052522-06ad8d41-4ce9-4a52-84f0-56b46cb6a572.png)
- large with badge
![image](https://user-images.githubusercontent.com/93647258/145052590-92fcbe78-c896-48d0-b2f6-f3decc944bf0.png)

steiningermona committed:
Merge branch 'badges-cy-109' into organization-menu-list-item-cy-217

steiningermona committed:
calculate text width based on badge width

steiningermona committed:
code cleanup

steiningermona committed:
update story to accept text

steiningermona committed:
Merge branch 'dev' into organization-menu-list-item-cy-217
